### PR TITLE
Link to search from AutoTextComplete

### PR DESCRIPTION
**Pull request by advaita13**
Issue : https://github.com/open-keychain/open-keychain/issues/1228

Added a small search button besides `EncryptKeyCompletionView`. Directly performs cloud key search using the text obtained from the view.

### CODE REVIEW

**Comment by advaita13** [comment]
@adithyaphilip Could you pls review and merge this.

File: OpenKeychain/src/main/java/org/sufficientlysecure/keychain/ui/EncryptModeAsymmetricFragment.java [hunk 1]
```diff
@@ -50,6 +52,7 @@
 
     private KeySpinner mSignKeySpinner;
     private EncryptKeyCompletionView mEncryptKeyView;
+    private String mQuery;
```
Comment (adithyaphilip) [inline OpenKeychain/src/main/java/org/sufficientlysecure/keychain/ui/EncryptModeAsymmetricFragment.java]: Why is this global? Try to get rid of all the warnings shown in Android Studio, that should prevent issues like this ("field can be converted to local").

**Comment by adithyaphilip** [comment]
This does let you search for missing keys, but it might be better to make the option part of the drop down itself (e.g. if there are no keys, the` EncryptKeyCompletionView` shows `Search for keys with  "John" online`). @Valodim your thoughts?

**Comment by Valodim** [comment]
I concur. The search option should be in the dropdown, when no key can be found for the typed text.

The KeyAdapter class already has some logic which is used in KeyListFragment to display the "You have no keys yet!" dummy item. It should be fairly simple to add a way to display an item based on the search query (which it already knows).

File: OpenKeychain/src/main/java/org/sufficientlysecure/keychain/ui/EncryptModeAsymmetricFragment.java (hunk 1 above)
Comment (advaita13) [inline OpenKeychain/src/main/java/org/sufficientlysecure/keychain/ui/EncryptModeAsymmetricFragment.java]: I wasn't getting any such warning.
Will make the change. Thanks.

**Comment by advaita13** [comment]
On it. Thanks guys.

**Comment by advaita13** [comment]
KeyListFragment has a different implementation of it altogether.
For this I will need to populate the list with `Search for keys with "John" online` and implement some sort of `OnClickListener` to handle search. Could you guys guide me a bit about how that drop-down menu is populated?

**Comment by Valodim** [comment]
The dropdown menu is populated from the database using a loader and adapter, also using a holder pattern.

**Comment by advaita13** [comment]
If the items in the dropdown menu are returned from the database directly then will it be possible to add a custom menu item?
Or how about using `swapCursor` to add a dummy cursor object on the dropdown?

**Comment by adithyaphilip** [comment]
Sure, go ahead and implement what you have in mind and we can take it from there. `swapCursor` seems to be the way to go. Also look at the `getViewForObject` function.

**Comment by advaita13** [comment]
I've been trying to get around this problem in multiple ways, no breakthrough so far. Here's why...
I created a MatrixCursor and added a row to it containing mostly null values to see if I can get it to show something on the drop down. Doesn't work. The dummy row (filled with mostly null values) isn't the right way to go since many other methods perform certain actions on the values and hence give an exception when those values are null.
As a result there is not way to pass off a dummy cursor without causing dependent functions to throw runtime errors.

The search button besides seems like the way to go for now. The button allows you to search for keys even beyond the query results displayed in the drop down.

**Comment by dschuermann** [comment]
Please open a new pull request if you have new implementation ideas. I will close this one for now.

**Pull request closed without merging**